File: model/Conta.cs
using System;

namespace SistemaBancario
{
    public abstract class Conta
    {
        protected int numero;
        protected Titular titular;
        protected decimal saldo;

        public Conta(int numero, Titular titular)
        {
            setNumero(numero);
            setTitular(titular);
            this.saldo = 0;
        }

        public Conta()
        {
        }

        public void setNumero(int numero)
        {
            if (numero <= 0)
            {
                throw new ArgumentException("O número da conta deve ser positivo.");
            }
            this.numero = numero;
        }

        public void setTitular(Titular titular)
        {
            if (titular == null)
            {
                throw new ArgumentException("O titular da conta não pode ser nulo.");
            }
            this.titular = titular;
        }

        public int getNumero()
        {
            return numero;
        }
        public Titular getTitular()
        {
            return titular;
        }

        public decimal getSaldo()
        {
            return saldo;
        }

        public abstract void Depositar(decimal valor);
        public abstract void Sacar(decimal valor);
        public override string ToString()
        {
            return $"Conta Número: {numero}, Titular: {titular.getNome()}, Saldo: {saldo:C}";
        }

        public void Transferir(Conta contaDestino, decimal valor)
        {
            if (contaDestino == null)
            {
                throw new ArgumentException("A conta de destino não pode ser nula.");
            }

            this.Sacar(valor);
            contaDestino.Depositar(valor);
        }
        
    }
}

File: services/Conta.php
<?php
require_once 'Titular.php';
abstract class Conta
{
    protected int $numero;
    protected Titular $titular;
    protected float $saldo;

    public function __construct(int $numero, Titular $titular)
    {
        $this->setNumero($numero);
        $this->titular = $titular;
        $this->saldo = 0.0;
    }

    public function setNumero(int $numero): void
    {
        if ($numero <= 0) {
            throw new Exception("O número da conta deve ser positivo.");
        }
        $this->numero = $numero;
    }

    public function getNumero(): int
    {
        return $this->numero;
    }

    public function getTitular(): Titular
    {
        return $this->titular;
    }

    public function getSaldo(): float
    {
        return $this->saldo;
    }

    abstract public function depositar(float $valor): void;
    
    abstract public function sacar(float $valor): void;

    public function __toString(): string
    {
        return sprintf(
            "Conta Número: %d, Titular: %s, Saldo: R$ %.2f",
            $this->numero,
            $this->titular->getNome(),
            $this->saldo
        );
    }

    public function transferir(Conta $contaDestino, float $valor): void
    {
        if ($valor <= 0) {
            throw new Exception("O valor da transferência deve ser positivo.");
        }

        if ($contaDestino === $this) {
            throw new Exception("Não é possível transferir para a mesma conta.");
        }


        $this->sacar($valor);
        $contaDestino->depositar($valor);
    }
}

?>